gssapi_kerberos.go

```diff
@@ -57,7 +57,7 @@ type KerberosClient interface {
 
 // writePackage appends length in big endian before the payload, and sends it to kafka
 func (krbAuth *GSSAPIKerberosAuth) writePackage(broker *Broker, payload []byte) (int, error) {
-	length := len(payload)
+	length := uint64(len(payload))
 	size := length + 4 // 4 byte length header + payload
 	if size > math.MaxUint32 {
```

Comment:
You might want to change this to `size > math.MaxInt32` rather than `MaxUint32`. Line 65 would panic on a 32-bit system if size is > `math.MaxInt32`. But, you are kinda asking for it if you are trying to write > 2GB of data all at once.